syntax = "proto3";

package bar;

enum MyEnum
{
    MY_ENUM_0 = 0;
    My_ENUM_1 = 1;
}

message MySmallMessage
{
    int64 val = 1;
}

message MyMessage
{
    double val = 1;
    string str = 2;
    MyEnum enm = 3;
    MySmallMessage mySmallMessage = 4;
    repeated MySmallMessage vec = 5;
    map<uint32, string> mp = 6;
}